src/plotman/configuration.py

```diff
@@ -1,9 +1,69 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+import desert
 import yaml
 
 
-def get_path():
-    return 'config.yaml'
+def get_validated_configs():
+    """Return a validated instance of the PlotmanConfig dataclass with data from config.yaml."""
+    config_path = "config.yaml"
+    schema = desert.schema(PlotmanConfig)
+    try:
+        with open(config_path, "r") as file:
+            config_file = yaml.load(file, Loader=yaml.SafeLoader)
+            return schema.load(config_file)
+    except FileNotFoundError:
+        print("No config.yaml file present in current working directory")
+
+
+# Data models used to deserializing/formatting config.yaml files.
+
+@dataclass
+class Archive:
+    rsyncd_module: str
+    rsyncd_path: str
+    rsyncd_bwlimit: int
+    rsyncd_host: str
+    rsyncd_user: str
+    index: Optional[int] = 0  # If not explicit, "index" will default to 0
+
+@dataclass
+class Directories:
+    log: str
+    tmp: List[str]
+    dst: List[str]
+    tmp2: Optional[str] = None
+    tmp_overrides: Optional[Dict[str, Dict[str, int]]] = None
```

Comment:
From my understanding, names of the keys in `tmp_overrides` are variable - as they should be the names of configurable directories. In the example config.yaml, they are `/mnt/tmp/00`, etc. 

From my understanding, `dataclasses` need consistent field naming

Reply:
Agreed, I was referring to the inner not the outer.  Let's make sure I'm getting my story straight here.

```yaml
directories:
        tmp_overrides:
                "/mnt/tmp/00":
                        tmpdir_max_jobs: 5
```

So something like this with another class defined for the override information.  Admittedly with just one attribute presently.

```python3
tmp_overrides: Optional[Dict[str, TemporaryDiskOverride]] = None
```

The outer dict stays with the keys being whatever temporary directory paths people specify.  This may just work or may require some tricks for desert/marshmallow.  I'm not certain.

Reply:
Changed 👍